File: sdk/go/plantquestsdk_accept_test.go

package plantquest

import (
  "testing"
)
         
func TestAssetAcceptLoad(t *testing.T) {
  endpoint := "http://localhost:4010/project01/plant01/stage01"
  apikey := "apikey"

  want := Data {
    Id: "string",
    Name: "string",
    Desc: "string",
  }

  var options Options
  options.Apikey = apikey
  options.Endpoint = endpoint
  client := Make(options)

  var data Data
  data.Id = "t01"

  asset, err := client.Asset().Load(data)
  if err != nil {
    t.Errorf("ERR: %q", err)
  }

  got := asset.Data
  
  // Todo: add check to all struct fields
  if got.Id != want.Id  {
    t.Errorf("Error getting asset accept load want %q and got %q", want.Id, got.Id)
  }

  if got.Name != want.Name  {
    t.Errorf("Error getting asset accept load want %q and got %q", want.Name, got.Name)
  }

  if got.Desc != want.Desc  {
    t.Errorf("Error getting asset accept load want %q and got %q", want.Desc, got.Desc)
  }
}

func TestGeofenceAcceptLoad(t *testing.T) {
  endpoint := "http://localhost:4010/project01/plant01/stage01"
  apikey := "apikey"

  want := Data {
    Id: "string",
    Name: "string",
    Desc: "string",
  }

  var options Options
  options.Apikey = apikey
  options.Endpoint = endpoint
  client := Make(options)

  var data Data
  data.Id = "t01"

  geofence, err := client.Geofence().Load(data)
  if err != nil {
    t.Errorf("ERR: %q", err)
  }

  got := geofence.Data
  
  // Todo: add check to all struct fields
  if got.Id != want.Id  {
    t.Errorf("Error getting geofence accept load want %q and got %q", want.Id, got.Id)
  }

  if got.Name != want.Name  {
    t.Errorf("Error getting geofence accept load want %q and got %q", want.Name, got.Name)
  }

  if got.Desc != want.Desc  {
    t.Errorf("Error getting geofence accept load want %q and got %q", want.Desc, got.Desc)
  }
}

func TestRoomAcceptLoad(t *testing.T) {
  endpoint := "http://localhost:4010/project01/plant01/stage01"
  apikey := "apikey"

  want := Data {
    Id: "string",
    Name: "string",
    Desc: "string",
  }

  var options Options
  options.Apikey = apikey
  options.Endpoint = endpoint
  client := Make(options)

  var data Data
  data.Id = "t01"

  room, err := client.Room().Load(data)
  if err != nil {
    t.Errorf("ERR: %q", err)
  }

  got := room.Data
  
  // Todo: add check to all struct fields
  if got.Id != want.Id  {
    t.Errorf("Error getting room accept load want %q and got %q", want.Id, got.Id)
  }

  if got.Name != want.Name  {
    t.Errorf("Error getting room accept load want %q and got %q", want.Name, got.Name)
  }

  if got.Desc != want.Desc  {
    t.Errorf("Error getting room accept load want %q and got %q", want.Desc, got.Desc)
  }
}

File: sdk/js/test/accept/plantquestSDK.accept.test.js

const { test, describe } = require('node:test')
const { equal, deepEqual } = require('node:assert')

const { PlantquestSDK } = require('../../')


describe('PlantquestSDK Acceptance Tests', ()=>{
  test('happy', async ()=>{
    const client = makeClient()
    const out = await client.Geofence().load({id:'gf01'})
    console.log('Geofence.load', out)
    equal(out.data.id,'gf01')
  })


  test('asset-load', async ()=>{
    const client = makeClient()
    const out = await client.Asset().load({id:'t01'})
    console.log('load', out)

    // equal(out.status, 200, 'Expected status code 200');

    // Define the expected format
    const expectedFormat = {
      id: 'string',
      name: 'string',
      desc: 'string',
      custom: 'object',
    };

    const keys = Object.keys(expectedFormat);
    keys.forEach((key) => {
      equal(typeof out.data[key], expectedFormat[key], 'Expected ' + key + ' to be a ' + expectedFormat[key]);
    })

  })


  test('geofence-load', async ()=>{
    const client = makeClient()
    const out = await client.Geofence().load({id:'t01'})
    console.log('load', out)

    // equal(out.status, 200, 'Expected status code 200');

    // Define the expected format
    const expectedFormat = {
      id: 'string',
      name: 'string',
      desc: 'string',
      custom: 'object',
    };

    const keys = Object.keys(expectedFormat);
    keys.forEach((key) => {
      equal(typeof out.data[key], expectedFormat[key], 'Expected ' + key + ' to be a ' + expectedFormat[key]);
    })

  })


  test('room-load', async ()=>{
    const client = makeClient()
    const out = await client.Room().load({id:'t01'})
    console.log('load', out)

    // equal(out.status, 200, 'Expected status code 200');

    // Define the expected format
    const expectedFormat = {
      id: 'string',
      name: 'string',
      desc: 'string',
      custom: 'object',
    };

    const keys = Object.keys(expectedFormat);
    keys.forEach((key) => {
      equal(typeof out.data[key], expectedFormat[key], 'Expected ' + key + ' to be a ' + expectedFormat[key]);
    })

  })


})


function makeClient(config) {
  const client = PlantquestSDK.make({
    endpoint: 'http://localhost:4010/project01/plant01/stage01',
    apikey: 'apikey',
    ...config
  })

  return client
}

